### ISSUE TITLE
Consider switching to `foldhash`

### ISSUE THREAD

**Issue by Centril** [clockworklabs/SpacetimeDB#1872]
Hashbrown recently switched to `foldhash` instead of `ahash` and maybe we should as well.

Perhaps we should split our hashmap/sets into 3 categories:
1. IntMap/Set -- identity hashing, when we don't need a hash at all, in the case of ids.
2. FastMap/Set -- using `foldhash-f` when we don't care about Hash DoS, e.g., when the data comes from the DB state and not from user input.
3. ResistMap/Set -- using `foldhash-q` or some other hasher like siphash.

This could gain us some performance wins.

**Comment by cloutiertyler** [comment]
I do not believe we expose our hash function anywhere outside the datastore. Moving to P3.
